build/cmake/utils/CMakeLists.txt

```diff
@@ -30,17 +30,18 @@ if(wxUSE_XRC)
         BUNDLE DESTINATION "bin"
         )
 
-    if(NOT WIN32_MSVC_NAMING)
+    if(NOT WIN32_MSVC_NAMING AND wxBUILD_INSTALL)
         if(IPHONE)
             set(EXE_SUFFIX ".app")
         else()
             set(EXE_SUFFIX ${CMAKE_EXECUTABLE_SUFFIX})
         endif()
 
-        wx_install(CODE "execute_process( \
+        # Don't use wx_install() here to preserve quoting.
+        install(CODE "execute_process( \
             COMMAND ${CMAKE_COMMAND} -E create_symlink \
             ${CMAKE_INSTALL_PREFIX}/bin/${wxrc_output_name}${EXE_SUFFIX} \
-            ${CMAKE_INSTALL_PREFIX}/bin/wxrc${EXE_SUFFIX} \
+            \"\$ENV{DESTDIR}${CMAKE_INSTALL_PREFIX}/bin/wxrc${EXE_SUFFIX}\" \
```

Comment:
I think for consistency we should use quotes around both, or none. Not just around one.

Reply:
So do you mean that _not_ quoting this line would still work? If so, let's not do this... I'd rather not add quotes around the otherwise unchanged line -- or, if we want to do it, do it everywhere and not just here. But see above about my understanding of CMake quoting.

Reply:
The quotes are escaped because they are used within other quotes: `install(CODE " ... \"quoted text\"  ... ")`. This is maybe less obvious because the command is split over multiple lines.

I added the quotes so it is possible to have spaces in the install path. for example the resulting code might look like: `cmake -E create_symlink "/test dir/bin/wxrc-3.2" "/test dir/bin/wxrc"`. Without quotes this will fail.
I've never tried spaces in the install path (or destdir) so there are probably other places that also should be quoted. So for now you could leave them out and it can be part of a PR that adds them everywhere where needed.

The variable `$ENV{DESTDIR}` is escaped so it will _not_ be evaluated during CMake's configure step, and it will appear as-is in the `cmake_install.cmake` script that is executed when installing.


Using `${ARGN}` in the `wx_install` macro seems to cause everything to be re-evaluated, removing the escape. Thats why I added the `wxBUILD_INSTALL` check and use `install` directly.
It is possible to keep using `wx_install` by escaping the escape: `\\\$ENV{DESTDIR}`

Reply:
OK, so for now I'll just remove the (escaped) quotes, right?

Reply:
yes

Reply:
I added the quotes in #22660